src/devices_models/devices/common/range_constraint.jl

```diff
@@ -14,7 +14,7 @@ function RangeConstraintInputsInternal(
         constraint_infos,
         constraint_name,
         variable_name,
-        nothing,
+        Vector{Symbol}(),
```

Comment:
Did you make this changes or are they repeat from @daniel-thom changes? it seems odd that this changes are here.

Reply:
What you see is the result of collaboration between Sourabh and me to incorporate his new range constraint struct with my refactor effort.

In my implementation there were either 0 or 1 bin variables.  Sourabh needs it to be 0, 1, or 3.  So, we made it an array.  An empty array replaces the 0 case.

Reply:
Ok thanks, it just seemed to me that this PR has too many changes in places where there shouldn't be any